src/firestore.rules.test.js

```diff
@@ -56,3 +61,38 @@ test('allows write to scholarships doc when signed in', () => {
     .set(newScholarship);
   return assertSucceeds(testDoc);
 });
+
+test('allows edit when you are author of scholarship', () => {
+  const testDoc = authedApp
+    .firestore()
+    .collection('scholarships')
+    .doc('KLJASDQW')
+    .set(newScholarship);
+
+  const updatedDoc = authedApp
+    .firestore()
+    .collection('scholarships')
+    .doc('KLJASDQW')
+    .set({ ...testDoc, name: 'updated name' });
+
+  return assertSucceeds(updatedDoc);
+});
+
+/*
+test('does not update scholarship when different user', () => {
+  const testDoc = authedApp
+    .firestore()
+    .collection('scholarships')
+    .doc('KLJASDQW')
+    .set(newScholarship);
+
+  testDoc.then((s) => console.log(s));
+
+  const updatedDoc = authedApp2
+    .firestore()
+    .collection('scholarships')
+    .doc('KLJASDQW')
+    .set({ ...testDoc, name: 'updated name' });
+
+  return assertFails(updatedDoc);
```

Comment:
I'm trying to figure out why this succeeds when it's suppose to fail 🤔

Reply:
My guess is that (like in the other test), you're not waiting for the `.set()` promise to complete so perhaps there's a race condition going on and the second promise succeeds because it checks against the `create` rule instead of the `update` one.

Try something like this:

```js
  await assertSucceeds(authedApp
    .firestore()
    .collection('scholarships')
    .doc('KLJASDQW')
    .set(newScholarship));

  return assertFails(authedApp2
    .firestore()
    .collection('scholarships')
    .doc('KLJASDQW')
    .set({ ...newScholarship, name: 'updated name' }));
```